src/core/searchResult/SearchResultBase.ts

```diff
@@ -80,10 +73,18 @@ export class SearchResultBase<T> implements SearchResult<T> {
     Reflect.defineProperty(this, '_response', {
       value: response
     });
-
-    this._controller = request.controller;
-    this._searchAction = 'search';
-    this._scrollAction = 'scroll';
+    Reflect.defineProperty(this, '_controller', {
+      value: request.controller,
+      writable: true
```

Comment:
Why are these writable?

Reply:
Because in the `SpecificationSearchResult` class we overload the controller with `collection`